File: Greedy/codeforces_1969B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1969/B


def main():
    t = int(input())
    for _ in range(t):
        s = str(input())

        ans = 0
        x = 0
        for char in s:
            if char == '1':
                x += 1
            else:
                if x > 0:
                    ans += (x + 1)
                x = 0
            
        print(ans)

if __name__ == "__main__":
    main()

File: Greedy/codeforces_1969B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1969/A
// submission link ===>> https://codeforces.com/contest/1969/submission/260259483

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        vector<ll> v(n);
        for (ll i = 0; i < n; i++)
        {
            cin >> v[i];
        }

        ll ans = 3;
        for (int i = 1; i <= n; i++){
            if(i==v[v[i-1]-1]){
                ans = 2;
                break;
            }
        }
        cout << ans << endl;
    }
    return 0;
}